### ISSUE TITLE
Push for non current rejected: force-with-lease should be offered

### ISSUE THREAD

**Issue by gerhardol** [gitextensions/gitextensions#8548]
## Feature description

If git-push fails, GE offers to pull-merge/rebase or push-force-with-lease if the current branch is rejected. This is the same if the current-only or multiple tab is used.

![push-current](https://user-images.githubusercontent.com/6248932/96348824-d8578b00-10ab-11eb-959b-ca38681829e1.JPG)

(ahead/behind in this screenshot, that is another PR)
![push-current-mult](https://user-images.githubusercontent.com/6248932/96348832-e3aab680-10ab-11eb-9b64-460b7ffbe722.JPG)

![push-failed](https://user-images.githubusercontent.com/6248932/96348838-f1f8d280-10ab-11eb-8774-4cd6279180a7.JPG)

![push-rejected-current](https://user-images.githubusercontent.com/6248932/96348843-fcb36780-10ab-11eb-948f-3c15df3810d9.JPG)

If any other branch is pushed, the push is just rejected. pull is not relevant for other than the current (not entirely, but it will be too complicated here).
Force box could be ticked (there are some limitations for force-with-lease per branch), but that could cause commits to be lost unintentionally.

The form should therefore offer force-with-lease also for non current branches.

![force-only](https://user-images.githubusercontent.com/6248932/96348994-fffb2300-10ac-11eb-8606-00085335c50c.JPG)

### Background

I often work with one or more official branches on top of master but other commits on top of that with non official patches.
When rebasing or applying fixup to "early" commits, I now just have to "rebase another branch" and push multiple branches (after #8515). The push requires force and I want that to be safer than the raw --force.

## Environment

GE master a573768a5b2abebf0515fd95ec35291978f09219, after #8515
